pages/map.py

```diff
@@ -244,33 +312,39 @@ def update_output(map_type, selected_user_ids, selected_user_emails, trips_data)
     Output('user-id-dropdown', 'disabled'),
     Output('user-email-dropdown', 'disabled'),
     Input('map-type-dropdown', 'value'),
+    Input('user-mode-dropdown', 'value'),
 )
-def control_user_dropdowns(map_type):
+def control_user_dropdowns(map_type,selected_user_modes):
     disabled = True
     if map_type == 'lines':
         disabled = False
+        if selected_user_modes:
+            disabled = True
     return disabled, disabled
 
 
+def process_trips_group(trips_group):
+    users_data = dict()
+    if trips_group:
+        keys = list(trips_group)
+        n = len(keys) % 360
+        k = 359 // (n - 1) if n > 1 else 0
+        for ind, key in enumerate(trips_group.groups.keys()):
+            color = f'hsl({ind * k}, 100%, 50%)'
+            trips = trips_group.get_group(key).sort_values('trip_start_time_str').to_dict("records")
```

Comment:
why do we need to sort the trips? Given a map-based visualization, there is effectively no ordering among trips.

Reply:
It is sorting the trip based on their start time, but as we are plotting all the co-ordinates collectively there is no advantage of sorting here. I will skip the sorting part.